susemanager-utils/susemanager-sls/src/states/appstreams.py

```diff
@@ -0,0 +1,101 @@
+'''
+Salt Custom State for configuring AppStreams
+
+This custom state provides functionality for configuring (enabling and disabling)
+AppStreams modules. Its 'configured' custom state offers a solution for managing
+AppStreams by facilitating both the enabling and disabling operations simultaneously.
+
+Note: When invoking the 'configured' state, the disable operation is executed first,
+ensuring a seamless transition and minimizing potential errors. This sequential approach
+is especially beneficial in scenarios where it's necessary to disable a previous version
+of the same AppStream module before enabling a newer version - In that case, the enabling
+operation would fail if the previous version were not disabled. Moreover, in the
+event of an error during the enable operation, the previously disabled modules are
+automatically reenabled.
+'''
+
+__virtualname__ = "appstreams"
+
+def __virtual__():
+    """
+    Only if the appstreams module is loaded
+    """
+    return (
+        __virtualname__
+        if ("appstreams.enable" in __salt__
+            and "appstreams.disable" in __salt__
+            and "appstreams.get_enabled_modules" in __salt__)
+        else (False, "Module appstreams: appstreams module can't be loaded")
+    )
+
+def configured(name, appstreams_to_enable, appstreams_to_disable):
```

Comment:
Like `configured` using passive voice, the arguments should also be named with a passive voice.
```
def configured(name, enabled_appstreams, disabled_appstreams):
```

Or, without repeating ourselves (it's `appstreams.configured` after all):
```
def configured(name, enabled, disabled):
```

Reply:
Actually, there is a conceptual issue: the method is supposed to update the selection of appstreams. So the `enabled` and `disabled` parameters do not reflect the state, but the changes to be made. So I think this method should be better written as an execution module.